server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/DefaultMetaRepository.java

```diff
@@ -238,6 +241,18 @@ static MirrorCredential findCredential(Iterable<MirrorCredential> credentials, U
             this.localRepo = requireNonNull(localRepo, "localRepo");
             this.localPath = firstNonNull(localPath, "/");
             this.remoteUri = requireNonNull(remoteUri, "remoteUri");
+            if (gitIgnore != null) {
+                if (gitIgnore instanceof Iterable) {
+                    this.gitIgnore = String.join("\n", (Iterable<String>) gitIgnore);
```

Comment:
There was one thing I missed.
Could you check that all elements in the `Iterable` are `String` before casting it to `Iterable<String>`?